stop_and_wait_plugin/src/stop_and_wait_plugin.cpp

```diff
@@ -43,432 +43,294 @@
+
+  lanelet::BasicPoint2d veh_pos(state.X_pos_global, state.Y_pos_global);
+  double starting_downtrack = wm_->routeTrackPos(veh_pos).downtrack;  // The vehicle position
+  double starting_speed = state.longitudinal_vel;
+
+  std::vector<lanelet::BasicPoint2d> route_points =
+      wm->sampleRoutePoints(std::min(starting_downtrack + 1, stop_and_wait_maneuver.end_dist),
+                            stop_and_wait_maneuver.end_dist, 1.0);  // TODO make step size a parameter
+  route_points.insert(route_points.begin(), veh_pos);
+
+  for (const auto& p : route_points)
+  {
+    PointSpeedPair pair;
+    pair.point = p;
+    pair.speed = starting_speed;
+    points_and_target_speeds.push_back(pair);
+  }
+
+  return points_and_target_speeds;
+}
+
+std::vector<cav_msgs::TrajectoryPlanPoint> StopandWait::trajectory_from_points_times_orientations(
+    const std::vector<lanelet::BasicPoint2d>& points, const std::vector<double>& times, const std::vector<double>& yaws,
+    ros::Time startTime)
+{
+  if (points.size() != times.size() || points.size() != yaws.size())
+  {
+    throw std::invalid_argument("All input vectors must have the same size");
+  }
+
+  std::vector<cav_msgs::TrajectoryPlanPoint> traj;
+  traj.reserve(points.size());
+
+  for (int i = 0; i < points.size(); i++)
+  {
+    cav_msgs::TrajectoryPlanPoint tpp;
+    ros::Duration relative_time(times[i]);
+    tpp.target_time = startTime + relative_time;
+    tpp.x = points[i].x();
+    tpp.y = points[i].y();
+    tpp.yaw = yaws[i];
+
+    tpp.controller_plugin_name = "default";
+    tpp.planner_plugin_name = plugin_discovery_msg_.name;
+
+    traj.push_back(tpp);
+  }
+
+  return traj;
+}
+// TODO how are we going to handle crawling to the stop line if stopped before?
+std::vector<cav_msgs::TrajectoryPlanPoint> StopandWait::compose_trajectory_from_centerline(
+    const std::vector<PointSpeedPair>& points, double starting_downtrack, double starting_speed, double stop_location,
+    double stop_location_buffer, ros::Time start_time)
+{
+  std::vector<cav_msgs::TrajectoryPlanPoint> plan;
+  if (points.size() == 0)
+  {
+    ROS_WARN_STREAM("No points to use as trajectory in stop and wait plugin");
+    return plan;
+  }
+
+  std::vector<PointSpeedPair> final_points;
+
+  double remaining_distance = stop_location - starting_downtrack; // TODO review. Should the remaining distance be in the middle of the stopping buffer?
+  double target_accel = config_.accel_limit_multiplier * config_.accel_limit;
+  double req_dist =
+      starting_speed / (2.0 * target_accel);  // Distance needed to go from current speed to 0 at target accel
+
+  if (req_dist > remaining_distance)
+  {
+    ROS_DEBUG_STREAM("Target Accel Update From: " << target_accel);
+    target_accel = starting_speed / (2.0 * remaining_distance);  // If we cannot reach the end point it the required
+                                                                 // distance update the accel target
```

Comment:
Adding a comment as a reminder to consider max vehicle acceleration. But based on discussion it can probably be ignored to let the vehicle do the best it can